### PR TITLE
fix: pointer ignored, performance improvements

### PR DESCRIPTION
**Pull request by lgalabru**
This PR is updating the config format:

```toml

    [storage]
    working_dir = "/hirosystems/ordhook/data"

    # The Http Api allows you to register / deregister
    # dynamically predicates.
    # Disable by default.
    #
    [http_api]
    http_port = 20456
    
    [network]
    mode = "mainnet"
    bitcoind_zmq_url = "tcp://0.0.0.0:18543"
    bitcoind_rpc_url = "http://0.0.0.0:8332"
    bitcoind_rpc_username = "xxx"
    bitcoind_rpc_password = "xxx"
    
    [snapshot]
    download_url = "https://archive.hiro.so/mainnet/ordhook/mainnet-ordhook-sqlite-latest"

    [resources]
    ulimit = 3096
    cpu_core_available = 12
    memory_available = 32
    bitcoind_rpc_threads = 4
    bitcoind_rpc_timeout = 15
    expected_observers_count = 1
```

### CODE REVIEW

File: components/ordhook-cli/src/config/file.rs [hunk 1]
```diff
@@ -1,29 +1,24 @@
...
 #[derive(Deserialize, Debug, Clone)]
 pub struct ConfigFile {
     pub storage: StorageConfigFile,
     pub http_api: Option<PredicatesApiConfigFile>,
-    pub limits: LimitsConfigFile,
+    pub resources: ResourcesConfigFile,
     pub network: NetworkConfigFile,
     pub logs: Option<LogConfigFile>,
-    pub bootstrap: Option<BootstrapConfigFile>,
+    pub snapthot: Option<SnapshotConfigFile>,
```
Comment (GuutBoy) [inline components/ordhook-cli/src/config/file.rs]: I wonder if this should be `snapshot` instead of `snapthot`?

Comment (lgalabru) [reply components/ordhook-cli/src/config/file.rs]: Thanks for catching this @GuutBoy! Addressed with https://github.com/hirosystems/ordhook/pull/250/commits/eca79f0e92ae82843f55d2e270a02874ad4e1120

File: components/ordhook-cli/src/config/file.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ impl ConfigFile {
             _ => return Err("network.mode not supported".to_string()),
         };
 
-        let snapshot = match config_file.snapthot {
+        let snapshot = match config_file.snapshot {
             Some(bootstrap) => match bootstrap.download_url {
```
Comment (GuutBoy) [inline components/ordhook-cli/src/config/file.rs:53]: Does this need to be changed to `snapshot` rather than `bootstrap` too? Sorry, I dont know Rust

Comment (lgalabru) [reply components/ordhook-cli/src/config/file.rs:53]: Correct!

Comment (lgalabru) [reply components/ordhook-cli/src/config/file.rs:53]: ah you mean the inner field? that's not required

Comment (GuutBoy) [reply components/ordhook-cli/src/config/file.rs:53]: haha, idk, I just changed it all to snapshot as the bootstrap word looked like it was being phased out. I seems like it is working here.

**Comment by GuutBoy** [comment]
Sorry to be difficult but now it just cant build this anymore. `cargo ordhook-intall` gives a bunch of errors like 
```
relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
```

**Comment by GuutBoy** [comment]
The above seems to be due to the tcmalloc thing. Rolling back to the typo fixing commit solves it.

**Comment by tapstoone** [comment]
Any progress ?

**Comment by lgalabru** [comment]
@0xbitfly yep, we're almost there!

**Pull request merged**